File: app/src/main/java/cn/bproject/neteasynews/adapter/VideoListAdapter.java
package cn.bproject.neteasynews.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.aspsine.irecyclerview.IViewHolder;
import com.bumptech.glide.Glide;

import java.util.ArrayList;

import cn.bproject.neteasynews.R;
import cn.bproject.neteasynews.bean.VideoBean;

/**
 * Created by liaozhoubei on 2017/1/7.
 */

public class VideoListAdapter extends RecyclerView.Adapter<VideoListAdapter.ViewHolder> {

    private Context mContext;
    private ArrayList<VideoBean> mVideoBeanList;
    private OnItemClickListener mOnItemClickListener;

    public VideoListAdapter(Context context, ArrayList<VideoBean> mVideoBeanList) {
        mContext = context;
        this.mVideoBeanList = mVideoBeanList;

    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = View.inflate(mContext, R.layout.item_video_layout, null);
        final ViewHolder holder = new ViewHolder(view);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final int position = holder.getIAdapterPosition();
                final VideoBean videoBean = mVideoBeanList.get(position);
                if (mOnItemClickListener != null) {
                    mOnItemClickListener.onItemClick(position, videoBean, v);
                }
            }
        });
        return holder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        final  VideoBean videoBean = mVideoBeanList.get(position);

        String imageSrc = videoBean.getCover();
        String title = videoBean.getTitle();
        String source = videoBean.getVideosource();
        String postTime = videoBean.getPtime();

        holder.tv_title.setText(title);
        holder.tv_from.setText(source);
        holder.tv_play_time.setText(postTime);

        holder.iv_video.setScaleType(ImageView.ScaleType.FIT_XY);
        Glide.with(mContext)
                .load(imageSrc)
                .placeholder(R.drawable.defaultbg_h)
                .crossFade()
                .into(holder.iv_video);

    }

    @Override
    public int getItemCount() {
        return mVideoBeanList.size();
    }

    /**
     * 设置Item点击监听
     *
     * @param listener
     */
    public void setOnItemClickListener(OnItemClickListener listener) {
        this.mOnItemClickListener = listener;
    }


    public interface OnItemClickListener<T> {
        void onItemClick(int position, T t, View v);
    }

    public class ViewHolder extends IViewHolder {

        public TextView tv_from ;
        public TextView tv_title ;
        public TextView tv_play_time ;
        public ImageView iv_video ;


        public ViewHolder(View itemView) {
            super(itemView);
            tv_from = (TextView) itemView.findViewById(R.id.tv_from);
            tv_title = (TextView) itemView.findViewById(R.id.tv_title);
            iv_video = (ImageView) itemView.findViewById(R.id.iv_video);
            tv_play_time = (TextView) itemView.findViewById(R.id.tv_play_time);

        }
    }


}

File: app/src/main/java/cn/bproject/neteasynews/adapter/VideoListAdapter.kt
package cn.bproject.neteasynews.adapter


import android.content.Intent
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView

import android.widget.TextView
import androidx.paging.PagedListAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import cn.bproject.neteasynews.R
import cn.bproject.neteasynews.activity.VideoDetailActivity

import cn.bproject.neteasynews.bean.VideoBean
import com.bumptech.glide.Glide

class VideoListAdapter : PagedListAdapter<VideoBean, VideoListAdapter.ViewHolder>(REPO_COMPARATOR) {

    private val VID = "VID"
    private val MP4URL = "MP4URL"
    private val BIG_IMG = 0
    private val SMALL_IMG = 1
    private val THREE_IMG = 2
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VideoListAdapter.ViewHolder {
        var view: View = View.inflate(parent.context, R.layout.item_video_layout, null)

        return ViewHolder(view)
    }



    override fun onBindViewHolder(holder: VideoListAdapter.ViewHolder, position: Int) {
        val videoBean = getItem(position)

        val imageSrc = videoBean?.cover
        val title = videoBean?.title
        val source = videoBean?.videosource
        val postTime = videoBean?.ptime

        holder.tv_title.text = title
        holder.tv_from.text = source
        holder.tv_play_time.text = postTime


        holder.iv_video.setScaleType(ImageView.ScaleType.FIT_XY)
        Glide.with(holder.iv_video.context)
            .load(imageSrc)
            .placeholder(R.drawable.defaultbg_h)
            .into(holder.iv_video)

        holder.iv_video.setOnClickListener(){
            val intent = Intent(holder.iv_video.context, VideoDetailActivity::class.java)
            intent.putExtra(VID, getItem(position)?.vid)
            intent.putExtra(MP4URL, getItem(position)?.mp4_url)
            holder.iv_video.context.startActivity(intent)
        }
    }

    private fun setNetPicture(url: String?, img: ImageView) {
        Glide.with(img.context)
            .load(url)
            .placeholder(R.drawable.defaultbg)
            .into(img)
    }

    companion object {
        private val REPO_COMPARATOR = object : DiffUtil.ItemCallback<VideoBean>() {
            override fun areItemsTheSame(oldItem: VideoBean, newItem: VideoBean): Boolean =
                oldItem.title == newItem.title

            override fun areContentsTheSame(oldItem: VideoBean, newItem: VideoBean): Boolean =
                oldItem.title == newItem.title
        }
    }

    class ViewHolder(var view: View) : RecyclerView.ViewHolder(view) {

        var tv_from: TextView = view.findViewById(R.id.tv_from);
        var tv_title: TextView = view.findViewById(R.id.tv_title);
        var tv_play_time: TextView = view.findViewById(R.id.tv_play_time);
        var iv_video: ImageView = view.findViewById(R.id.iv_video);
    }
}